include/logging/log_backend.h

```diff
@@ -28,6 +28,12 @@ struct log_backend;
 struct log_backend_api {
 	void (*put)(const struct log_backend *const backend,
 		    struct log_msg *msg);
+	void (*sync_msg)(const struct log_backend *const backend,
```

Comment:
on the other hand if we do the switching then we will need to compile in sync handling functions so that will increase logger subsystem size. In sync mode, panic has no impact, system will continue to work as usual (to some extent panic = sync mode)